### DISCUSSION TITLE
linux 环境下是否可以使用NCW模式，是什么原因导致的

### DISCUSSION THREAD

**Discussion by chengzenqiang** [CefView/QCefView#409] [General]
请问下，为什么linux 在NCW无效， 我自己写了一个demo 只是简单在qt界面上嵌入cef.发现 页面访问成功，但是无法显示
```cpp
#include "mainwindow.h"
#include "ui_mainwindow.h"
#include "simple_handler.h"

MainWindow::MainWindow(QWidget *parent)
: QMainWindow(parent)
, ui(new Ui::MainWindow)
{
ui->setupUi(this);

}

MainWindow::~MainWindow()
{
delete ui;
}

void MainWindow::on_pushButton_2_clicked()
{
// ui->gridLayout->addWidget(widget);
// SimpleHandler implements browser-level callbacks.
CefRefPtr handler(new SimpleHandler(false));
// Specify CEF browser settings here.
CefBrowserSettings browser_settings;

QString url="http://www.baidu.com";
CefWindowInfo window_info;

CefBrowserSettings browserSettings;
window_info.SetAsChild((CefWindowHandle)ui->label->winId(), { 0, 0, ui->label->width(), ui->label->height()});
CefBrowserHost::CreateBrowser(window_info,        // window info
                                             handler,           // handler
                                             url.toStdString(), // url
                                             browserSettings,   // settings
                                             nullptr,
                                             CefRequestContext::GetGlobalContext());
}
```
---------------------------------------------------------------------------------------------
![image](https://github.com/CefView/QCefView/assets/32353345/f1bb19fa-5714-4dad-a89c-5ca22158b03b)

**Comment by chengzenqiang** [comment]
我用的官方handler部分用的是官方cefsimple示例。已经解决了初始化崩溃报错问题。但是现在就是无法显示网页，感觉挺像windows在qt界面上设置 透明属性，或者没有给exe设置权限的问题。但是在linux 环境下，开发经验比较少，请大佬帮忙解答下这方面的困惑

**Reply by tishion** [reply]
linux 这一块你可以请教@L-Super 来帮忙看看，他非常熟悉。

**Reply by L-Super** [reply]
> 我用的官方handler部分用的是官方cefsimple示例。已经解决了初始化崩溃报错问题。但是现在就是无法显示网页，感觉挺像windows在qt界面上设置 透明属性，或者没有给exe设置权限的问题。但是在linux 环境下，开发经验比较少，请大佬帮忙解答下这方面的困惑

如果你准备使用Qt + CEF，建议你查阅相关资料，就可以避免使用QLabel作为view容器来使用的错误。如果你准备使用QCefView，那么你将避免遇到这些问题。

**Reply by chengzenqiang** [reply]
我打算用Qt+CEF 去做， 请问下 @L-Super 大神在哪里可以查看到详细的资料
我之前都是在这几个网站查看资料，是否有补充
官方网址：https://bitbucket.org/chromiumembedded/cef/src/master/
GitHub镜像：https://github.com/chromiumembedded/cef
自动构建版本下载：https://cef-builds.spotifycdn.com/index.html
Cef wiki：https://bitbucket.org/chromiumembedded/cef/wiki/Home
官方论坛：https://magpcss.org/ceforum/

**Reply by L-Super** [reply]
> 我打算用Qt+CEF 去做， 请问下 @L-Super 大神在哪里可以查看到详细的资料 我之前都是在这几个网站查看资料，是否有补充 官方网址：https://bitbucket.org/chromiumembedded/cef/src/master/ GitHub镜像：https://github.com/chromiumembedded/cef 自动构建版本下载：https://cef-builds.spotifycdn.com/index.html Cef wiki：https://bitbucket.org/chromiumembedded/cef/wiki/Home 官方论坛：https://magpcss.org/ceforum/

这似乎超出了QCefView的范畴
